src/storage/mod.rs

```diff
@@ -1444,6 +1376,182 @@ impl<E: Engine, L: LockManager> Storage<E, L> {
     }
 }
 
+enum RawStore<S: Snapshot> {
```

Comment:
This file is too large.... We shall move this code into another independent file

Reply:
will do